src/Cake.Common/Tools/DotNetCore/Build/DotNetCoreBuildSettings.cs

```diff
@@ -29,6 +29,11 @@ public sealed class DotNetCoreBuildSettings : DotNetCoreSettings
         public string Configuration { get; set; }
```

Comment:
@adamhathcock please in this PR remove `string Configuration { get; set; }` and it's usages in favor of the new `ICollection<string> Configurations { get; set; }`

We've discussed internally with in @cake-build/team even though it's a breaking change, to keep API clean and consistent, and as we feel we catch this early.